bookwyrm/models/activitypub_mixin.py

```diff
@@ -554,6 +555,16 @@ async def sign_and_send(
                 logger.exception(
                     "Failed to send broadcast to %s: %s", destination, response.reason
                 )
+                logger.info("Trying again with legacy keyId")
+                # try with incorrect keyId to enable communication with legacy Bookwyrm servers
+                legacy_signature = make_signature("post", sender, destination, now, digest, True)
+                headers["Signature"] = legacy_signature
+                async with session.post(destination, data=data, headers=headers) as response: 
```

Comment:
I think it would be cleaner to make a `sign_and_send` take `use_legacy_key` and call it recursively, to avoid duplicating some of this code.